src/test/kotlin/org/rust/ide/intentions/IfLetToMatchIntentionTest.kt

```diff
@@ -636,4 +636,47 @@ class IfLetToMatchIntentionTest : RsIntentionTestBase(IfLetToMatchIntention()) {
             }
         }
     """)
+
+    fun `test irrefutable pattern`() = doAvailableTest("""
+        struct Id(u32);
+        struct S { a: Id, b: u32 }
+        fn foo(s: S) {
+            if let S { a: Id(ref name), .. } = s/*caret*/ {
+                let _x = a;
```

Comment:
```suggestion
                let _x = name;
```

Otherwise, this code cannot be compiled because `a` here is not binding
Fix it please in other tests as well